### PR TITLE
Conform with thesis submission guidelines

### PR DESCRIPTION
**Pull request by AlexKuhnle**
Changes based on the thesis submission guidelines: https://www.cambridgestudents.cam.ac.uk/your-course/examinations/graduate-exam-information/after-examination/degree-approval-and-1

- Swapped order of abstract and declaration:  titlepage - declaration - abstract - acknowledgements
- Added `official` document option which:
    (a) adds title and author to the abstract page
    (b) adds deposit&copying declaration form as first page (can be replaced with scanned filled-in pdf)

### CODE REVIEW

**Comment by urbas** [comment]
The PDF failed to build with this error:
```
! Undefined control sequence.
l.133 \ifcam
            @official
? 
! Emergency stop.
l.133 \ifcam
            @official
!  ==> Fatal error occurred, no output PDF file produced!
```

**Comment by urbas** [comment]
Please add a test for your new option to [`.circleci/tests`](https://github.com/cambridge/thesis/tree/master/.circleci/tests).

**Comment by urbas** [comment]
Thanks a lot @AlexKuhnle for your contribution!

**Comment by AlexKuhnle** [comment]
Okay, I will have a look and add a test. I compiled my thesis that way, so I thought it would work, but maybe I missed a line somewhere. Sorry for not testing properly.

**Comment by urbas** [comment]
@AlexKuhnle: no worries. Can you rebase on top of the latest `master` branch? I updated the build image to Ubuntu 18.04 LTS (it was 16.04 before). This image comes with newer LaTeX packages. Perhaps that could fix your build.

**Comment by urbas** [comment]
@AlexKuhnle: Could you add the link you mentioned in the PR description to [the answer to Q2 in the README.md](https://github.com/cambridge/thesis/#q2-where-can-i-find-the-thesis-formatting-guidelines-this-class-is-based-on)? This link: https://www.cambridgestudents.cam.ac.uk/your-course/examinations/graduate-exam-information/after-examination/degree-approval-and-1

**Comment by AlexKuhnle** [comment]
So, I think this should all be done and fixed now. Let me know if there are any other problems.

File: thesis.tex [hunk 1]
```diff
@@ -124,6 +125,20 @@
 %%
 \begin{document}
 
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Insert "Deposit & Copying of Dissertation Declaration" form/scan or comment.
+%%
+\ifcam@official
+\includepdf{declaration.pdf}%
```
[the remark this answers is not in the document]
Comment (AlexKuhnle) [reply thesis.tex]: Okay, makes sense.

File: cam-thesis.cls [hunk 2]
```diff
@@ -520,6 +511,33 @@ This dissertation does not exceed the prescribed limit of 60\,000 words.
...
+\fi
+
+\ifcam@firstyr
+% First yr report doesn't need a standalone abstract
+\else
+
+\chapter*{Abstract}
+\thispagestyle{empty}
+
+\ifcam@official
```
[the remark this answers is not in the document]
Comment (AlexKuhnle) [reply cam-thesis.cls]: Done.

File: cam-thesis.cls [hunk 3]
```diff
@@ -64,6 +64,10 @@
 \newif\ifcam@techreport\cam@techreportfalse
 \DeclareOption{techreport}{\cam@techreporttrue}
 
+% official - formats the thesis according to the official submission guidelines.
+\newif\ifcam@official\cam@officialfalse
```
[the remark this answers is not in the document]
Comment (AlexKuhnle) [reply cam-thesis.cls]: Okay, happy to make it default.

**Comment by AlexKuhnle** [comment]
Addressed your comments.

File: .circleci/tests/official/thesis.tex [hunk 4]
```diff
@@ -0,0 +1,109 @@
+\documentclass[official,withindex,glossary]{cam-thesis}
```
Comment (urbas) [inline .circleci/tests/official/thesis.tex]: Could you also remove this sample now that the `official` option is no longer there?

File: README.md [hunk 5]
```diff
@@ -174,17 +174,21 @@ You can also mail
...
 
-The University of Cambridge guidelines:
+The University of Cambridge submission guidelines:
 
-> [http://www.admin.cam.ac.uk/students/studentregistry/exams/submission/phd/format.html](http://www.admin.cam.ac.uk/students/studentregistry/exams/submission/phd/format.html)
+> [https://www.cambridgestudents.cam.ac.uk/your-course/examinations/graduate-exam-information/submitting-and-examination/phd-msc-mlitt/submit](https://www.cambridgestudents.cam.ac.uk/your-course/examinations/graduate-exam-information/submitting-and-examination/phd-msc-mlitt/submit)
+
+The University of Cambridge final submission guidelines:
+
+> [https://www.cambridgestudents.cam.ac.uk/your-course/examinations/graduate-exam-information/after-examination/degree-approval-and-1](https://www.cambridgestudents.cam.ac.uk/your-course/examinations/graduate-exam-information/after-examination/degree-approval-and-1)
```
Comment (urbas) [inline README.md]: Great! Thanks for improving these references!

File: .circleci/tests/official/thesis.tex (hunk 4 above)
Comment (AlexKuhnle) [reply .circleci/tests/official/thesis.tex]: You're right, removed the test case now.

**Comment by urbas** [comment]
Awesome, thanks @AlexKuhnle for the contribution! :+1: 

Let's get this fella into master :shipit:

**Pull request merged**